src/nunit.analyzers.tests/EqualToIncompatibleTypes/EqualToIncompatibleTypesAnalyzerTests.cs

```diff
@@ -1091,5 +1091,66 @@ public AnotherRecord(string name)
             RoslynAssert.Valid(analyzer, testCode);
         }
 #endif
+
+        [TestCase("1.1f")]
+        [TestCase("2.2f")]
+        public void AnalyzeIsNotNaNOnValidTypes(string actual)
+        {
+            var testCode = TestUtility.WrapInTestMethod(@$"
+                var actual = {actual};
+                Assert.That(actual, Is.Not.NaN);
+            ");
+
+            RoslynAssert.Valid(analyzer, testCode);
+        }
+
+        [TestCase("1.1f")]
+        [TestCase("2.2f")]
+        public void AnalyzeIsNaNOnValidTypes(string actual)
+        {
+            var testCode = TestUtility.WrapInTestMethod(@$"
+                var actual = {actual};
+                Assert.That(actual, Is.NaN);
+            ");
+
+            RoslynAssert.Valid(analyzer, testCode);
+        }
+
+        [TestCase("3.3m")]
+        [TestCase("4")]
+        [TestCase("string.Empty")]
+        public void AnalyzeIsNotNaNOnUnsupportedTypes(string actual)
+        {
+            var testCode = TestUtility.WrapInTestMethod(@$"
+                var actual = {actual};
+                Assert.That(actual, Is.Not.NaN);
+            ");
+
+            RoslynAssert.Diagnostics(analyzer, expectedDiagnostic, testCode);
+        }
+
+        [TestCase("3.3m")]
+        [TestCase("4")]
+        [TestCase("string.Empty")]
+        public void AnalyzeIsNaNOnUnsupportedTypes(string actual)
+        {
+            var testCode = TestUtility.WrapInTestMethod(@$"
+                var actual = {actual};
+                Assert.That(actual, Is.Positive.And.↓Not.NaN);
```

Comment:
The test name `AnalyzeIsNaNOnUnsupportedTypes` says `IsNan` - so should we remove the `Not` from the assert